### ISSUE TITLE
Missing files are skipped, but only indicated through message on stderr

### ISSUE THREAD

**Issue by michaelforney** [madler/pigz#43]
I noticed that when pigz is given a file name that doesn't exist, it skips the file, printing

```
pigz: skipping: foo does not exist
```

and exits with status 0. I can imagine some uses for this behavior, but it seems more useful to me to treat this an error and exit non-0.

Was the current behavior chosen to handle some common usage pattern? If not, what are your thoughts on changing pigz to exit non-0 in this case? It looks like gzip exits with status 1 after printing an error message.

**Comment by madler** [comment]
I will make it compatible with gzip.

**Comment by jhart-r7** [comment]
Was this ever addressed?  This behavior changed in 2.3.2 and represents (IMO) a big change in how `pigz` operates for files that don't exist as well as files that do exist but aren't compressed:

`pigz` 2.3.3:

```
$  rm -f blah.gz missing.gz; pigz --version; pigz -t missing.gz; echo $?; pigz -dc missing.gz > /dev/null; echo $?; touch blah.gz; pigz -t blah.gz; echo $?; pigz -dc blah.gz > /dev/null; echo $?
pigz 2.3.3
pigz: skipping: missing.gz does not exist
0
pigz: skipping: missing.gz does not exist
0
0
0
```

`pigz` 2.3:

```
$ rm -f blah.gz missing.gz; pigz --version; pigz -t missing.gz; echo $?; pigz -dc missing.gz > /dev/null; echo $?; touch blah.gz; pigz -t blah.gz; echo $?; pigz -dc blah.gz > /dev/null; echo $?
pigz 2.3
pigz: missing.gz does not exist -- skipping
2
pigz: missing.gz does not exist -- skipping
2
0
0
```

https://github.com/madler/pigz/commit/0c967f989d2f1afe1829cf7cf2f6d36846a625ec looks like the relevant commit, which I realize was a long time ago.  Several linux distributions are still shipping 2.3 (Ubuntu 14.04, for example).  

Would you accept a PR to perhaps make this behavior configurable?  Or perhaps revert back to the original mode of exiting non-zero in situations like this by default and making it an option to ignore errors like this?

This new behavior makes at least one recent use case for me harder.  I've got a file that I fetch that is supposed to be compressed but isn't always for reasons I can't control.  So with 2.3, I can `pigz -t` the file beforehand and just `cat` it rather than `pigz -dc`'ing it.  But with 2.3.2+, I can't do that, and I suspect this breaks all sorts of workflows.

Thanks in advance!

**Comment by madler** [comment]
pigz currently returns 1 if an input does not exist.
